pkgs/by-name/op/openimagedenoise/package.nix

```diff
@@ -15,19 +15,20 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "openimagedenoise";
-  version = "2.2.2";
+  version = "2.3.2";
 
   # The release tarballs include pretrained weights, which would otherwise need to be fetched with git-lfs
   src = fetchzip {
-    url = "https://github.com/OpenImageDenoise/oidn/releases/download/v${finalAttrs.version}/oidn-${finalAttrs.version}.src.tar.gz";
-    sha256 = "sha256-ZIrs4oEb+PzdMh2x2BUFXKyu/HBlFb3CJX24ciEHy3Q=";
+    url = "https://github.com/RenderKit/oidn/releases/download/v${finalAttrs.version}/oidn-${finalAttrs.version}.src.tar.gz";
+    sha256 = "sha256-yTa6U/1idfidbfNTQ7mXcroe7M4eM7Frxi45A/7e2A8=";
   };
 
   patches = lib.optional cudaSupport ./cuda.patch;
 
   postPatch = ''
-    substituteInPlace devices/metal/CMakeLists.txt \
-      --replace-fail "AppleClang" "Clang"
+    # fix build failure with GCC14
+    substituteInPlace cmake/oidn_platform.cmake \
+      --replace-fail "set(CMAKE_CXX_STANDARD 11)" "set(CMAKE_CXX_STANDARD 14)"
```

Comment:
Looks like one of the patches that won't be compatible with `overrideAttrs (_: { src = ...; })` and that we'd have to keep updating over time. We should maybe warn [upstream](https://github.com/RenderKit/oidn/commit/21bbc89947c0ac504d5daa91e594a6651bebf5f2#diff-1d4d3fb431cc405ec3a543dab305a6d63c60ac453774e781888ebb2f2b997e00R10-R29) that the meaning of cxx_standard has apparently changed from gcc13 to gcc14? Also I don't think it makes much sense to set `CMAKE_CXX_STANDARD` over `target_compile_features` or `CXX_STANDARD`. I'm not a cmake expert, but I feel like `CMAKE_*` variants are better set on the cli or in toolchain files

Reply:
I opened the issue yesterday https://github.com/RenderKit/oidn/issues/256